### PR TITLE
Feature/kr/setidentity backend

### PR DESCRIPTION
**Pull request by KanishkaRajputd**
### Summary by CodeRabbit

- New Feature: Introduced on-chain identity verification process with multiple steps including amount breakdown, identity form, and social verification.
- New Feature: Added email and Twitter verification handlers in the API.
- New Feature: Implemented a new component `OnChainIdentity` for managing on-chain identity settings.
- New Feature: Enhanced `AppLayout` to include an option for setting on-chain identity in the user dropdown menu.
- New Feature: Updated `Home` component to check identity verification status and render an `OnChainIdentity` component based on the verification status.
- New Feature: Added new components `SocialVerification`, `TotalAmountBreakdown`, `VerificationSuccessScreen` for handling different aspects of the on-chain identity verification process.
- Refactor: Updated `.eslintrc.json` to disable the "indent" rule.
- Refactor: Modified `getTwitterCallback` function to return the Twitter handle instead of a static success message.
- Refactor: Updated `formatedBalance` function to control the number of decimal places in the formatted balance.

### Summary by CodeRabbit

Release Notes:

- New Feature: Added authentication functionality using Twitter as a provider in a Next.js application.
- New Feature: Introduced API handler functions for verifying user identities through email and Twitter.
- Bug Fix: Modified the `onPostWritten` function to include additional logic related to indexing posts in Algolia.
- Refactor: Made changes to import statements, component rendering, and useEffect hooks in `pages/_app.tsx`.
- Style: Several changes to the `Gov2Home` component in `pages/opengov/index.tsx`.
- Chore: Updated dependencies in `.eslintrc.json`, `cloud-functions/package.json`, and `package.json`.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
 		"arrow-spacing":["warn",{ "before": true, "after": true }],
 		"comma-dangle": ["error", "never"],
 		"header/header": [2, "./header.js"],
-		"indent": ["error", "tab"],
+		"indent": "off",
```
Comment (alphainfinitus) [inline .eslintrc.json:43]: please undo this change

File: cloud-functions/package.json [hunk 2]
```diff
@@ -8,18 +8,23 @@
     "shell": "npm run build && firebase functions:shell",
     "start": "npm run shell",
     "deploy": "firebase deploy --only functions",
-    "logs": "firebase functions:log"
+    "logs": "firebase functions:log",
+    "server": "npm run build -- --watch | firebase emulators:start --only functions"
```
Comment (alphainfinitus) [inline cloud-functions/package.json:12]: please remove `server` script

File: package.json [hunk 3]
```diff
@@ -46,6 +46,8 @@
     "bn.js": "^5.2.1",
     "chart.js": "^4.2.1",
     "classnames": "^2.3.2",
+    "cookie": "^0.5.0",
```
Comment (alphainfinitus) [inline package.json:49]: please check if these packages are needed.
I don't think they're being used

File: pages/api/v1/verification/judgement-call.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply pages/api/v1/verification/judgement-call.ts:3]: false positive

File: pages/api/v1/verification/judgement-call.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply pages/api/v1/verification/judgement-call.ts:4]: :clown_face:

File: pages/api/v1/verification/index.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+if (apiKey) {
+	sgMail.setApiKey(apiKey);
+}
+
+const firestore = firebaseAdmin.firestore();
+
+const handler: NextApiHandler<IVerificationResponse | MessageType> = async (req, res) => {
+	const { type, checkingVerified } = req.query as unknown as IReq;
+	const { account, userId } = req.body;
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply pages/api/v1/verification/index.ts:46]: @KanishkaRajputd

File: pages/api/v1/verification/index.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+if (apiKey) {
+	sgMail.setApiKey(apiKey);
+}
+
+const firestore = firebaseAdmin.firestore();
+
+const handler: NextApiHandler<IVerificationResponse | MessageType> = async (req, res) => {
+	const { type, checkingVerified } = req.query as unknown as IReq;
```
Comment (alphainfinitus) [inline pages/api/v1/verification/index.ts:45]: why do we have data accepted in both query and params ?
also if you require data in `req.body` implement an early check for request type to be POST only

File: pages/api/v1/verification/index.ts [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	const { account, userId } = req.body;
+	const token = getTokenFromReq(req);
+
+	const user = await authServiceInstance.GetUser(token);
+
+	if(!userId || !user?.id) throw apiErrorWithStatusCode(messages.UNAUTHORISED, 403);
```
Comment (alphainfinitus) [inline pages/api/v1/verification/index.ts:51]: why is handler throwing an error ?

File: pages/api/v1/verification/onchain-identity-via-polkassembly.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import { NextApiHandler } from 'next';
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import messages from '~src/auth/utils/messages';
+import { firestore_db } from '~src/services/firebaseInit';
+import getTokenFromReq from '~src/auth/utils/getTokenFromReq';
+import authServiceInstance from '~src/auth/auth';
+import { MessageType } from '~src/auth/types';
+
+const handler: NextApiHandler<MessageType | string> = async (req, res) => {
```
[the remark this answers is not in the document]
Comment (KanishkaRajputd) [reply pages/api/v1/verification/onchain-identity-via-polkassembly.ts:14]: false

**Review by AleemAlam — APPROVED** [review]

**Pull request merged**